### ISSUE TITLE
Support for Mac OS X

### ISSUE THREAD

**Issue by mitar** [sandstorm-io/meteor-spk#17]
Please support Mac OS X.

**Comment by kentonv** [comment]
Sorry but this isn't directly possible with the current design. `meteor-spk` calls the `spk` tool, which is part of Sandstorm, which runs on Linux. The `dev` command in particular needs to talk to your local Sandstorm server and does all kinds of Linux-specific stuff.

We have some thoughts about a completely different tool design where the app could actually run in dev mode remotely, possibly even on a different OS. But, this is a long ways off.

**Comment by mitar** [comment]
Hm, just to convert a Meteor app you need `spk`?

**Comment by kentonv** [comment]
Yes, you need the `spk` tool to build `spk`s (Sandstorm package files)...

**Comment by mitar** [comment]
Aha. I wanted just to convert Meteor app and then user Travis CI for building `spk` on every push to `master` branch of my app. Is this not a good idea?

**Comment by kentonv** [comment]
Arguably for the specific purpose of `spk pack` (not `spk dev`), it shouldn't be Linux-dependent. We might be able to make that much work on OSX. Could you file a bug against Sandstorm asking that basic `spk` commands (i.e. not `dev`) be made to work on other platforms? Note that it might still require some cleanup on our part since a lot of our code simply assumes Linux.

**Comment by kentonv** [comment]
Although it occurs to me that _usually_ `spk dev` cannot work on non-Linux because it pulls in binaries and libraries from you local system and expects those to be Linux. It's really only in the case of `meteor-spk` that this makes any sense, because `meteor-spk` just so happens to come with a complete binary/library bundle so that binaries from the system aren't needed...

**Comment by mitar** [comment]
> Although it occurs to me that usually spk dev cannot work on non-Linux because it pulls in binaries and libraries from you local system and expects those to be Linux.

I would be OK if that has to be some GNU tools I have to install with `brew` as a requirement.

**Comment by kentonv** [comment]
> I would be OK if that has to be some GNU tools I have to install with brew as a requirement.

Sorry, that won't work, because these binaries are being put into the package, and the package ultimately runs on Linux (inside Sandstorm).

**Comment by mitar** [comment]
Could it be made so that it uses `vagrant-spk`?

**Comment by zarvox** [comment]
There is already a meteor stack for `vagrant-spk`.  For example, the Rocket.Chat developers have packaged Rocket.Chat from an OSX machine with `vagrant-spk`.

**Comment by mitar** [comment]
Huh, that would be great to be more known. Especially above @kentonv gave a feeling that this is not possible. At that time I didn't know of existence of `vagrant-spk`. I Googled "Meteor Sandstorm" and got to this package. It would be great to have some resources/references how to do it from Mac OS X.
